akka-actor/src/main/scala/akka/actor/AbstractProps.scala

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.actor
+
+import java.lang.reflect.{ Modifier, ParameterizedType, TypeVariable }
+
+import akka.dispatch._
+import akka.japi.Creator
+import akka.routing._
+import akka.util.Reflect
+
+import scala.annotation.varargs
+import scala.language.existentials
+import scala.reflect.ClassTag
+import akka.japi.Util.immutableSeq
+
+/**
+ * Java API: Factory for Props instances.
+ */
+trait AbstractProps {
```

Comment:
I also believe that code separation jvm dependant / agnostic framework is a good design just in style with this great project.
More projects can benefits from this in the future.